src/components/common/ColonyActions/CountDownTimer/helpers.ts

```diff
@@ -0,0 +1,197 @@
+    case MotionState.Staking:
+    case MotionState.Staked:
+    case MotionState.Objection:
+      return motionTimeoutPeriods.timeLeftToStake;
+    case MotionState.Voting:
+      return motionTimeoutPeriods.timeLeftToVote;
+    case MotionState.Reveal:
+      return motionTimeoutPeriods.timeLeftToReveal;
+    case MotionState.Escalation:
+      return motionTimeoutPeriods.timeLeftToEscalate;
+    default:
+      return '-1';
+  }
+};
+
+const useMotionTimeoutPeriods = (colonyAddress = '', motionId: number) => {
+  const { data, loading, refetch } = useGetMotionTimeoutPeriodsQuery({
+    variables: { input: { colonyAddress, motionId } },
+    notifyOnNetworkStatusChange: true,
+    fetchPolicy: 'network-only',
+  });
+  const motionTimeoutPeriods = data?.getMotionTimeoutPeriods || {
+    timeLeftToEscalate: '-1',
+    timeLeftToVote: '-1',
+    timeLeftToReveal: '-1',
+    timeLeftToStake: '-1',
+  };
+
+  return {
+    motionTimeoutPeriods,
+    loadingMotionTimeoutPeriods: loading,
+    refetchMotionTimeoutPeriods: refetch,
+  };
+};
+
+export const useMotionCountdown = (
+  state: MotionState,
+  motionId: number,
+  refetchMotionState: RefetchMotionState,
+) => {
+  const { colony } = useColonyContext();
+  const { user } = useAppContext();
+  const dispatch = useDispatch();
+  const {
+    motionStakes: { percentage: percentageStaked },
+  } = useStakingWidgetContext();
+  const isFullyNayStaked = percentageStaked.nay === '100';
+
+  const {
+    motionTimeoutPeriods,
+    loadingMotionTimeoutPeriods,
+    refetchMotionTimeoutPeriods,
+  } = useMotionTimeoutPeriods(colony?.colonyAddress, motionId);
+
+  const currentStatePeriodInMs = getCurrentStatePeriodInMs(
+    state,
+    motionTimeoutPeriods,
+  );
+
+  const [timeLeft, setTimeLeft] = useState<number>(-1);
+
+  const prevStateRef: MutableRefObject<MotionState | null> = useRef(null);
+  const isStakingPhaseState =
+    state === MotionState.Staking ||
+    state === MotionState.Staked ||
+    state === MotionState.Objection;
+
+  /*
+   * Set the initial timeout
+   *
+   * @NOTE The extra 5 seconds are to account for the time in between the blockchain
+   * timeout being hit and the next block being processed.
+   *
+   * If there are no blockes being processes between the time the motion "should" finish
+   * and the time we refresh the state (because we are basically keeping a parallel time count)
+   * the state of the motion won't change.
+   *
+   * This is confusing UX and might need revisiting in the future with some clever messages
+   * for the user to let them know what's going on
+   *
+   * So we are "faking" this by adding an extra 5 to the alloted time, so that the
+   * blockchain has time to process any transactions, detect the motion's timeout,
+   * and change the state so we can refresh it.
+   */
+  useEffect(() => {
+    if (
+      !loadingMotionTimeoutPeriods &&
+      /*  @NOTE: If we have loaded the motion timeout periods, we should only set/reset the timer:
```

Comment:
The comment says:

> If we have loaded the motion timeout periods

But the check is:

`!loadingMotionTimeoutPeriods`

i.e. not loaded the motion timeout periods

Or am I misunderstanding?

Reply:
I got here because I was trying to debug the following:

Let one motion hit zero, and fast forward time so it actually updates the motion state.

Create a new motion. The timeLeft state is zero.

Reply:
The way I see it, when we get to this point in the flow we should have already requested the motion timeout periods via `useMotionTimeoutPeriods` above this. So, if `!loadingMotionTimeoutPeriods` is `true` then this would mean that either the query finished fetching the data or it tried to and there was an error, both ways the motion timeout periods have stopped loading and we should have some values to initialize the timer. (Even if the query fails, as we have fallbacks)

Reply:
Ah yes thanks, I was reading that wrong